### PR TITLE
Fix tests for the Config and DI components

### PR DESCRIPTION
**Pull request by derrabus**

### CODE REVIEW

File: src/Symfony/Component/Config/Definition/ArrayNode.php [hunk 1]
```diff
@@ -48,7 +48,7 @@ public function setNormalizeKeys($normalizeKeys)
      *
      * @param mixed $value
      *
-     * @return array The value with normalized keys
+     * @return mixed The value with normalized keys
```
Comment (derrabus) [inline src/Symfony/Component/Config/Definition/ArrayNode.php:51]: if `$value` is not an array, it is returned immediately.

File: src/Symfony/Component/Config/Util/XmlUtils.php [hunk 2]
```diff
@@ -152,7 +152,7 @@ public static function loadFile($file, $schemaOrCallable = null)
      * @param \DOMElement $element     A \DOMElement instance
      * @param bool        $checkPrefix Check prefix in an element or an attribute name
      *
-     * @return array A PHP array
+     * @return mixed
```
Comment (derrabus) [inline src/Symfony/Component/Config/Util/XmlUtils.php:155]: This method can return virtually any non-object type and we have tests covering this. That is a bit surprising, given that the method name ends with `ToArray`.

File: src/Symfony/Component/DependencyInjection/Extension/ExtensionInterface.php [hunk 3]
```diff
@@ -37,7 +37,7 @@ public function getNamespace();
     /**
      * Returns the base path for the XSD files.
      *
-     * @return string The XSD base path
+     * @return string|false
```
Comment (derrabus) [inline src/Symfony/Component/DependencyInjection/Extension/ExtensionInterface.php:40]: 😢

Comment (nicolas-grekas) [reply src/Symfony/Component/DependencyInjection/Extension/ExtensionInterface.php:40]: let's fix that with a deprecation, same as the guard method

**Pull request merged**